react/card.js

```diff
@@ -2,36 +2,24 @@ import React, { Component } from 'react'
 import PropTypes from 'prop-types'
 import { intlShape, injectIntl } from 'react-intl'
 
+import { fields } from './constants'
 import { Input, Button } from 'vtex.styleguide'
 
-const fields = [
-  'relationName',
-  'emissionDate',
-  'expiringDate',
-  'caption',
-  'redemptionCode',
-  'userdocument',
-  'conditions',
-  'balance',
-  'group',
-  'active',
-]
-
 class Card extends Component {
-  constructor() {
-    super()
+  constructor(props) {
+    super(props)
 
     this.state = {
-      relationName: 'relation Diego17',
-      emissionDate: '2018-10-04T17:57:50.743Z',
-      expiringDate: '2019-10-04T17:57:50.743Z',
-      caption: 'GiftCardTestKevin2',
-      redemptionCode: 'redemptioncodetest',
-      userdocument: '80344263428',
-      conditions: '{"Document":"80344263428"}',
-      balance: 171,
-      group: 'group1',
-      active: 'true',
+      relationName: props.cardData.relationName || '',
```

Comment:
não seria melhor ter default props com todos os dados ao inves disso?

Reply:
Pode ser mas acho que basicamente vai botar isso tudo "em baixo" só.

Reply:
Mas de qualquer forma ficaria sintaticamente melhor escrito, pq no this.state vc quer que as props entrem, e as props podem vir do `defaultProps` ou de fato de fora do componente.

Reply:
Na vdd esse `this.state` recebendo as props é inútil, pq vc só precisaria fazer as `defaultProps` com as propriedades vazias e referenciar o `this.props.cardData` no componente que ele não quebraria